go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -297,27 +292,25 @@ func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall
 	return updateErrors
 }
 
-func (oc *Controller) deleteEgressFirewall(egressFirewall *egressfirewallapi.EgressFirewall, txn *util.NBTxn) error {
-	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
+func (oc *Controller) deleteEgressFirewall(egressFirewallObj *egressfirewallapi.EgressFirewall, txn *util.NBTxn) error {
+	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewallObj.Name, egressFirewallObj.Namespace)
 	deleteDNS := false
+	obj, _ := oc.egressFirewalls.LoadAndDelete(egressFirewallObj.Namespace)
```

Comment:
it would probably be prudent to check the return value for the bool here otherwise casting on the next line would result in a crash if nil is returned

Reply:
check the boolean return value